sdk/eventhub/event-hubs/src/batchingReceiver.ts

```diff
@@ -57,8 +62,13 @@ export class BatchingReceiver extends EventHubReceiver {
       let onSessionClose: OnAmqpEvent;
       let waitTimer: any;
       let actionAfterWaitTimeout: Func<void, void>;
+      let aborter: Aborter;
+      let onAborted: () => {};
       // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
       const finalAction = (timeOver: boolean) => {
+        if (aborter) {
+          aborter.removeEventListener("abort", onAborted);
```

Comment:
We need to remove the listener in 2 other places from where we reject the promise. This is similar to what we did for the `send` request if you recall.

All paths through which the operation completes (resolved or rejected promise), should remove the listener